src/main/java/ru/skypro/homework/controller/AdsController.java

```diff
@@ -51,89 +67,105 @@ public Ads getAllAds() {
         return adsService.getAllAds();
     }
 
-    @Operation(summary = "Добавление объявления")
-    @ApiResponses({
-            @ApiResponse(responseCode = "201", description = "Created", content = {
-                    @Content(mediaType = MediaType.APPLICATION_JSON_VALUE,
-                            schema = @Schema(implementation = AdView.class))}),
-            @ApiResponse(responseCode = "401", description = "Unauthorized", content = {
-                    @Content(mediaType = "")})})
-    @PostMapping(consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
-    public ResponseEntity<AdDTO> createAd(@RequestPart(name = "properties") @Valid CreateOrUpdateAd properties,
-                                          @RequestPart(name = "image") MultipartFile image) {
-        return ResponseEntity.ok(adsService.createAd(properties, image)); //TODO
-    }
-
 
+    /**
+     * Этот метод обрабатывает GET-запросы на получение информации о конкретном объявлении
+     *
+     * @param id идентификатор объявления
+     * @return Возвращает всю информацию об объявлении с указанным id
+     */
     @GetMapping("/{id}")
     @Operation(summary = "Получение информации об объявлении")
     @ApiResponses({
             @ApiResponse(responseCode = "200", description = "Ok", content = {
                     @Content(mediaType = MediaType.APPLICATION_JSON_VALUE,
                             schema = @Schema(implementation = ExtendedAd.class))}),
-            @ApiResponse(responseCode = "401", description = "Unauthorized", content = {
-                    @Content(mediaType = "")}),
-            @ApiResponse(responseCode = "404", description = "Not found", content = {
-                    @Content(mediaType = "")})})
+            @ApiResponse(responseCode = "401", description = "Unauthorized", content = @Content),
+            @ApiResponse(responseCode = "404", description = "Not found", content = @Content)})
     public ResponseEntity<ExtendedAd> getAdFullInfo(@PathVariable("id") Integer id) {
         return ResponseEntity.ok(adsService.getAdFullInfo(id));
     }
 
+    /**
+     * Этот метод обрабатывает DELETE-запросы на удаление объявления
+     *
+     * @param id идентификатор объявления
+     * @param authentication данные авторизированного пользователя
+     */
     @DeleteMapping("/{id}")
     @Operation(summary = "Удаление объявления")
     @ApiResponses({
             @ApiResponse(responseCode = "204", description = "No Content"),
             @ApiResponse(responseCode = "401", description = "Unauthorized"),
             @ApiResponse(responseCode = "403", description = "Forbidden"),
             @ApiResponse(responseCode = "404", description = "Not found")})
-    public void deleteAd(@PathVariable("id") Integer id) {
+    @PreAuthorize("hasRole('ADMIN') or hasAuthority(@adsServiceImpl.getAdFullInfo(#id).getEmail().equalsIgnoreCase(authentication.name))")
```

Comment:
а зачем во втором условии authority проверяется? Должно быть:
```suggestion
    @PreAuthorize("hasRole('ADMIN') or @adsServiceImpl.getAdFullInfo(#id).getEmail().equalsIgnoreCase(authentication.name)")
```